homeassistant/components/lyric/climate.py

```diff
@@ -0,0 +1,317 @@
+        for location in self.coordinator.data.locations:
+            for device in location.devices:
+                if device.macID == self._device.macID:
+                    return (
+                        device.maxHeatSetpoint
+                        if "Heat" in device.allowedModes
+                        else device.maxCoolSetpoint
+                    )
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+        target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+
+        for location in self.coordinator.data.locations:
+            for device in location.devices:
+                if device.macID == self._device.macID:
+                    if device.hasDualSetpointStatus:
+                        if target_temp_low is not None and target_temp_high is not None:
+                            temp = (target_temp_low, target_temp_high)
+                    else:
+                        temp = kwargs.get(ATTR_TEMPERATURE)
+                    _LOGGER.debug("Set temperature: %s", temp)
+                    try:
+                        await self.coordinator.data.update_thermostat(
+                            location, device, heatSetpoint=temp
+                        )
+                    except (
+                        LyricAuthenticationException,
+                        LyricException,
+                        ClientResponseError,
+                    ) as exception:
+                        _LOGGER.error(exception)
+        await self.coordinator.async_refresh()
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set hvac mode."""
+        for location in self.coordinator.data.locations:
+            for device in location.devices:
+                if device.macID == self._device.macID:
+                    _LOGGER.debug("Set hvac mode: %s", hvac_mode)
+                    try:
+                        await self.coordinator.data.update_thermostat(
+                            location, device, mode=LYRIC_HVAC_MODES[hvac_mode]
+                        )
+                    except (
+                        LyricAuthenticationException,
+                        LyricException,
+                        ClientResponseError,
+                    ) as exception:
+                        _LOGGER.error(exception)
+        await self.coordinator.async_refresh()
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set preset (PermanentHold, HoldUntil, NoHold, VacationHold) mode."""
+        for location in self.coordinator.data.locations:
+            for device in location.devices:
+                if device.macID == self._device.macID:
+                    _LOGGER.debug("Set preset mode: %s", preset_mode)
+                    try:
+                        await self.coordinator.data.update_thermostat(
```

Comment:
Pass in the object and save it in the instance instead of calling the coordinator